# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [12]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [13]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2         V3         V4        V5        V6        V7        
1 0.1079335 0.04369473 0.64290581 0.5320323 0.7250710 0.4460170 0.77450875
2 0.1803311 0.54293061 0.05915424 0.5425978 0.4679879 0.4576827 0.01377805
  V8        V9        V10       V11        V12       V13        V14       
1 0.4836622 0.6665157 0.4088902 0.07327765 0.6569648 0.05253621 0.06263485
2 0.1283199 0.8023867 0.4640010 0.45805685 0.6590157 0.04067860 0.57977398
  V15       V16        V17       V18       V19       V20       
1 0.4084187 0.02262569 0.5669418 0.7930036 0.1046416 0.75559900
2 0.4786330 0.14584459 0.1475763 0.3274080 0.3487920 0.07041281

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [22]:
df2 <- df1 %>%
gather(Key, Value)  
head(df2)

Key Value     
1 V1  0.10793347
2 V1  0.18033107
3 V2  0.04369473
4 V2  0.54293061
5 V3  0.64290581
6 V3  0.05915424

## Assignment 2

We will generate some data:

In [15]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [16]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [17]:
df5 <- full_join(df3,df4,by="id")
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook